scripts/irods/test/test_setting_user_password.py

```diff
@@ -0,0 +1,1088 @@
+		user_session.assert_icommand("ils", "STDOUT", user_session.home_collection)
+
+	# Run iexit to remove all the credentials files. We accept output on stdout here because the test is likely
+	# running as the service account Linux user and this causes iexit to print a warning which does not apply here.
+	user_session.assert_icommand(["iexit", "--remove-session-token-file", "-f"], "STDOUT")
+
+
+@unittest.skipIf(test.settings.TOPOLOGY_FROM_RESOURCE_SERVER, "Must configure catalog provider for these tests.")
+class test_modifying_user_password(unittest.TestCase):
+	@classmethod
+	def setUpClass(self):
+		# Create an admin user / session for all the tests so we don't have to keep making it for each test.
+		self.admin = session.mkuser_and_return_session("rodsadmin", "otherrods", "rods", lib.get_hostname())
+
+		# Make a backup of the server_config and configure insecure mode for irods authentication scheme if the tests
+		# are not being run with TLS enabled.
+		if False == test.settings.USE_SSL:
+			self.server_config_backup = tempfile.NamedTemporaryFile(prefix=os.path.basename(paths.server_config_path())).name
+			shutil.copyfile(paths.server_config_path(), self.server_config_backup)
+			with open(paths.server_config_path()) as f:
+				server_config = json.load(f)
+			server_config['plugin_configuration']['authentication']['irods'] = {
+				'insecure_mode': True
+			}
+			lib.update_json_file_from_dict(paths.server_config_path(), server_config)
+			IrodsController().reload_configuration()
+
+	@classmethod
+	def tearDownClass(self):
+		# Restore the original server configuration if necessary.
+		if False == test.settings.USE_SSL:
+			shutil.copyfile(self.server_config_backup, paths.server_config_path())
+			IrodsController().reload_configuration()
+
+		# Exit the test admin session and remove the user.
+		self.admin.__exit__()
+		with session.make_session_for_existing_admin() as admin_session:
+			admin_session.assert_icommand(["iadmin", "rmuser", self.admin.username])
+
+	def setUp(self):
+		# Make a test user without a password. The method being called here will not authenticate the user because
+		# there will not be a password set. This ensures that the user has no password set in any table.
+		self.test_user = session.mkuser_and_return_session("rodsuser", "bilbo", None, lib.get_hostname())
+
+	def tearDown(self):
+		# Exit the test user session and remove the user.
+		self.test_user.__exit__()
+		self.admin.assert_icommand(["iadmin", "rmuser", self.test_user.username])
+
+	def test_config_legacy_sets_password_for_native_scheme_but_not_irods_scheme(self):
+		config = IrodsConfig()
+		with lib.file_backed_up(config.server_config_path):
+			# Update server config and reload configuration.
+			config.server_config["user_password_storage_mode"] = "legacy"
+			lib.update_json_file_from_dict(config.server_config_path, config.server_config)
+			IrodsController(config).reload_configuration()
```

Comment:
Should we reset the server's password storage mode on class teardown? Does that happen already?

I'm thinking about what happens if a test fails and does not restore the password storage mode. Will this affect unrelated tests?